bin/database.py

```diff
@@ -1024,6 +1025,32 @@ def upgrade_to_28():
         config.db.sessions.update({'_id': session['_id']}, session)
 
 
+
+
+def upgrade_to_29_closure(user):
+
+    avatars = user['avatars']
+    if avatars.get('custom') and not 'https:' in avatars['custom']:
+        if user['avatar'] == user['avatars']['custom']:
+            config.db.users.update_one({'_id': user['_id']},
+                {'$set': {'avatar': user['avatars'].get('provider', None)}}
+            )
+        logging.info('Deleting custom ...')
+        config.db.users.update_one({'_id': user['_id']},
+            {'$unset': {"avatars.custom": ""}}
+        )
+    return True
+
+
+def upgrade_to_29():
+    """
+    Enforces HTTPS urls for user avatars
+    """
+    
+    users = config.db.users.find({})
```

Comment:
I just thought of one other thing: Since we always try to limit the number of items we have to iterate over in db updates, you could modify the `find` to only return users that have `avatar.custom` set.